### DISCUSSION TITLE
To have a controll on raw answers from LLMs before being parsed in doc.ents in NER tasks

### DISCUSSION THREAD

**Discussion by innocent-charles** [explosion/spacy-llm#385] [Ideas]
Hello! I was suggesting if it could be possible to also allow people to have the access control or be able to use the raw answers in NER returned by the LLMs just before being passed in the doc.ents. 

Why ? 
Because i have faced situations like LLMs return answer that are semantically correct but they are not as they have been written in the document. So this has caused error when those responses being taken to doc.ents . 

Example : 
i was extracting date of birth in the document : in the document it has been written as 20/11/1993 , when i check logs the answer was 
month: November
day: 20th
year: 1993 

which semantically, LLM has got it correct .......

But with this spacy LLM after final pipeline of doc.ents -- it seems that it can't give out those entities, 

so was wandering on that.

**Comment by rmitsch** [comment]
Hi @innocent-charles, sometimes the reply provided by the LLM doesn't fit the data schema requested in the prompt. In that case parsing fails and we can't extract/map the data automatically. If you want raw access to the data, you can set `save_io` to `True`:
```ini
[components.llm]
factory = "llm"
save_io = True

```

**Comment by innocent-charles** [comment]
Then , where and how do i read response after save_io=True ?

**Reply by rmitsch** [reply]
In `Doc.user_data["llm_io"]`. E. g.
```python
doc = nlp("This is a test")
print(doc.user_data["llm_io"]["prompt"])  # Output raw prompt
print(doc.user_data["llm_io"]["response"])  # Output raw response
```

**Comment by innocent-charles** [comment]
Thanks a lot, Raphel, I have got it.
